pkg/matrix/matrix.go

```diff
@@ -184,6 +184,14 @@ func (me *Matrix) RemoveMetric(key string) {
 	delete(me.metrics, key)
 }
 
+func (me *Matrix) RemoveExceptMetric(key string) {
+	for k := range me.metrics {
```

Comment:
instead of iterating through the metrics, deleting all but one, can we create a new map with only the one we want to keep. O(1) vs O(n)

Reply:
this method only touches `metrics`, no need to create a new matrix or if there is this method should do it too. I meant something like this, which is O(1)
```
func (me *Matrix) RemoveExceptMetric(key string) {
	prev, ok := me.metrics[key]
	if !ok {
		return
	}
	me.metrics = make(map[string]Metric)
	me.metrics[key] = prev
}
```